crates/accelerate/src/commutation_checker.rs

```diff
@@ -725,6 +721,31 @@ pub enum CommutationLibraryEntry {
     QubitMapping(HashMap<SmallVec<[Option<Qubit>; 2]>, bool>),
 }
 
+impl<'py> IntoPyObject<'py> for CommutationLibraryEntry {
+    type Target = PyAny;
+    type Output = Bound<'py, Self::Target>;
+    type Error = PyErr;
+
+    fn into_pyobject(self, py: Python<'py>) -> Result<Self::Output, Self::Error> {
+        let py_out = match self {
+            CommutationLibraryEntry::Commutes(b) => {
+                let temp = b.into_pyobject(py)?;
+                <pyo3::Bound<'_, PyBool> as Clone>::clone(&temp).into_any()
+            }
```

Comment:
I think this would be the expected way to do this. The impl of `IntoPyObject` for `bool` gives us a `Borrowed<PyBool>`, and then by calling trait method `BoundObject::into_bound` on that, we get an owned `Bound<PyBool>`.

```suggestion
            CommutationLibraryEntry::Commutes(b) => b.into_pyobject(py)?.into_bound().into_any(),
```